frontend/src/components/SearchIssues/SearchIssues.tsx

```diff
@@ -0,0 +1,94 @@
+import { Form } from '@highlight-run/ui/components'
+import { Select } from 'antd'
+import { useState } from 'react'
+import React from 'react'
+
+import { useSearchIssuesLazyQuery } from '@/graph/generated/hooks'
+import { IntegrationType } from '@/graph/generated/schemas'
+import { useDebouncedValue } from '@/hooks/useDebouncedValue'
+import { IssueTrackerIntegration } from '@/pages/IntegrationsPage/IssueTrackerIntegrations'
+
+import styles from './SearchIssues.module.css'
+export interface SearchOption {
+	value: string
+	label: string
+	id: string
+	url: string
+}
+
+export interface SearchIssuesProps {
+	onSelect: (option: SearchOption) => void
+	integration: IssueTrackerIntegration
+	project_id: string
+}
+
+export const SearchIssues = ({
+	onSelect,
+	integration,
+	project_id,
+}: SearchIssuesProps) => {
+	const [selectedOption, setSelectOption] = React.useState<
+		SearchOption | undefined
+	>({
+		label: '',
+		value: '',
+		id: '',
+		url: '',
+	})
+	const [query, setQuery] = useState<string>('')
+
+	const debouncedQuery = useDebouncedValue(query) || ''
+	const [searchIssues, { data, loading }] = useSearchIssuesLazyQuery()
+
+	React.useEffect(() => {
+		debouncedQuery &&
+			searchIssues({
+				variables: {
+					project_id,
+					query: debouncedQuery,
+					integration_type: integration.name as IntegrationType,
+				},
+				fetchPolicy: 'no-cache',
+			})
+	}, [searchIssues, project_id, debouncedQuery, integration])
+
+	const options = React.useMemo(() => {
+		return (
+			data?.search_issues.map((s) => ({
+				id: s.id,
+				url: s.issue_url,
+				value: s.issue_url,
+				label: s.title,
```

Comment:
This looks good to me, and maybe a future improvement - I assume there is not a standard way to include an identifier in the beginning of the label?

I'm thinking incase teams have duplicate or similar issues, it may be useful to show the id so they can help differentiate or identify which one to use. For example https://github.com/highlight/highlight/issues/7852 may be displayed as:
`[7852] Fix nested != queries`

Reply:
Some issue ids are quite long and will make the label harder to read

Reply:
I will consider using the issue ID prefix for those with short ids but that might make the ui a bit inconsistent?

Reply:
<img width="412" alt="Screenshot 2024-02-28 at 11 17 16 PM" src="https://github.com/highlight/highlight/assets/119384208/04d8987f-7f52-40fc-bcc0-3e7af0941f66">
<img width="478" alt="Screenshot 2024-02-28 at 11 27 46 PM" src="https://github.com/highlight/highlight/assets/119384208/1c1010be-ea03-47c4-863e-d098bb608172">
<img width="402" alt="Screenshot 2024-02-28 at 11 27 59 PM" src="https://github.com/highlight/highlight/assets/119384208/5a57bd32-e7b8-44a6-96d3-5695991c37c3">
<img width="435" alt="Screenshot 2024-02-28 at 11 28 11 PM" src="https://github.com/highlight/highlight/assets/119384208/6f0c6eea-b3a0-4fe5-8b22-c6a6f663712e">

I've updated these and I already have duplicate issues in one of the pictures that is now distinguishable
I try to make the issue number prefixes representative of how it is done on the respective issue integrations.

So most use "#" prefix while others just do their own thing like jira [Al-24]